.travis.yml

```diff
@@ -13,18 +13,24 @@ before_install:
   - if [ -e docker/previous_image.tar.gz ]; then docker load -i docker/previous_image.tar.gz; fi
 
 install:
+  - ci_env=`bash <(curl -s https://codecov.io/env)`
   - docker build -t iot-lab-gateway --cache-from=iot-lab-gateway .
   - docker build -t iot-lab-gateway-tests tests
 
 before_cache:
   - docker save -o docker/previous_image.tar.gz iot-lab-gateway
 
+after_success:
+  - docker run -v $PWD:/home/iot-lab-gateway $ci_env --privileged iot-lab-gateway-tests -e upload_coverage
+
 script:
   - >
-    docker run -v gateway_code:/home/iot-lab-gateway/gateway_code
-    -v $CWD/tests_utils/cfg_dir:/var/local/config
+    docker run
+    -v $PWD/tests_utils/cfg_dir:/var/local/config
+    -v $PWD:/home/iot-lab-gateway
     -e BOARD_TYPE=m3
     -e CONTROL_NODE_TYPE=no
     -e HOSTNAME=m3-00 
     -e IOTLAB_GATEWAY_NO_INTEGRATION_TESTS
+    $ci_env
```

Comment:
Is this still needed ?

Reply:
Yes, basically we're running codecov inside the docker container, it needs to know some things about the CI environment which are in TRAVIS_* environment variables, this is a shortcut to pass them all at once, see https://github.com/codecov/support/wiki/Testing-with-Docker

Reply:
ok so what is the point of the call to upload_coverage on success ?

Reply:
It's uploading the coverage to codecov.io when the tests are successful

Reply:
Yes, using codecov. I thing the codecov environement variables (from $ci_env) are only useful in this case.

Reply:
Oh now I see what you mean. You're right, this $ci_env shouldnt be needed in the docker run in script.